datafusion/src/physical_plan/functions.rs

```diff
@@ -3680,6 +3704,60 @@ mod tests {
             StringArray
         );
         #[cfg(feature = "unicode_expressions")]
+        test_function!(
```

Comment:
```sql
postgres=# select a from t;
    a
----------
 alphabet
(1 rows)

postgres=# select substring(a, 0, 5) from t;
 substring
-----------
 alph
(1 rows)

```

Reply:
Hi @ovr, Sorry, I think there are still some problems.

Reply:
You are right, i did another iteration and looks like I found a way how to get similar behaviour

PG:


```
select substr('alphabet', 0, 5) as s1, substr('alphabet', -5, 10) as s2, substr('alphabet', -5, 4) as s3;

[
  {
    "s1": "alph",
    "s2": "alph",
    "s3": ""
  }
]
```

DF:

```
❯ select substr('alphabet', 0, 5) as s1, substr('alphabet', -5, 10) as s2, substr('alphabet', -5, 4) as s3;
+------+------+----+
| s1   | s2   | s3 |
+------+------+----+
| alph | alph |    |
+------+------+----+
1 row in set. Query took 0.016 seconds.

```